src/test/java/org/dz/solution/task32/Task32Test.java

```diff
@@ -0,0 +1,174 @@
+        assertEquals("", actualResult);
+    }
+
+    @Test
+    public void case4() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1 3 4 6 13 14");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n5\n7\n8\n9\n10\n11\n12\n15\n16\n17\n18\n19\n20", actualResult);
+    }
+
+    @Test
+    public void case5() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "1");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19\n20", actualResult);
+    }
+
+    @Test
+    public void case6() throws IOException {
+        Files.writeString(Path.of(SOURCE_DATA_FILE), "20");
+        calculateSkippedNumbers();
+        String actualResult = Files.readString(Path.of(RESULT_FILE));
+        assertEquals("1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16\n17\n18\n19", actualResult);
+    }
+
+    private void calculateSkippedNumbers() {
+        int countNumbers = findTheNumberOfNumbersInAFile(SOURCE_DATA_FILE);
+        int[] numbersFromFile = writeNumbersToArray(countNumbers, SOURCE_DATA_FILE);
+        int[] skippedNumbers = findTheMissingNumbers(numbersFromFile);
+        writeArrayToFile(skippedNumbers, RESULT_FILE);
+    }
+
+    private int findTheNumberOfNumbersInAFile(String sourceDataFile) {
+        int count = 0;
+        try (BufferedReader br = new BufferedReader(new FileReader(sourceDataFile))) {
+            String line = br.readLine();
+            if (line == null) {
+                return 0;
+            }
+            String[] words = line.split(WORD_DELIMITER);
+            for (int i = 0; i < words.length; i++) {
+                if (words.length > count) {
+                    count++;
+                }
+            }
+            System.out.println(count);
+        } catch (IOException e) {
+            throw new RuntimeException("Array creation failed", e);
+        }
+        return count;
+    }
+
+    private int[] writeNumbersToArray(int countNumbers, String sourceDataFile) {
+        int[] array;
+        try (BufferedReader br = new BufferedReader(new FileReader(sourceDataFile))) {
+            String line = br.readLine();
+            if (line == null) {
+                return new int[0];
+            }
+            String[] elements = line.split(" ");
+            array = new int[countNumbers];
+            for (int i = 0; i < elements.length; i++) {
+                array[i] = Integer.parseInt(elements[i]);
+                System.out.print(array[i] + " ");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Array creation failed", e);
+        }
+        return array;
+    }
+
+    private int[] findTheMissingNumbers(int[] numbersFromFile) {
+        int[] skippedNumbers = new int[20];
```

Comment:
Try to use ArrayList<Integer> instead of new int[20];

When skippedNumbers is an ArrayList the code will be simpler.

You need to remember how to use ArrayList from previous tasks.